.rubocop.yml

```diff
@@ -66,8 +66,59 @@ Layout/LineContinuationLeadingSpace: #`new in 1.31.1
 Layout/LineContinuationSpacing: #`new in 1.31.1
   Enabled: true
 
+Lint/AmbiguousAssignment: # new in 1.7
+  Enabled: true
+Lint/AmbiguousOperatorPrecedence: # new in 1.21
+  # We do this only in rare cases; Always doing it would add uneeded clutter
+  Enabled: false
+Lint/AmbiguousRange: # new in 1.19
+  Enabled: true
 # https://docs.rubocop.org/rubocop/cops_lint.html#lintconstantoverwritteninrescue
-Lint/ConstantOverwrittenInRescue: #`new in 1.31.1
+Lint/ConstantOverwrittenInRescue: # new in 1.31.1
+  Enabled: true
+Lint/DeprecatedConstants: # new in 1.8
+  Enabled: true
+Lint/DuplicateBranch: # new in 1.3
+  Enabled: true
+Lint/DuplicateRegexpCharacterClassElement: # new in 1.1
+  Enabled: true
+Lint/EmptyBlock: # new in 1.1
+  Enabled: true
+Lint/EmptyClass: # new in 1.3
+  Enabled: true
+Lint/EmptyInPattern: # new in 1.16
+  Enabled: true
+Lint/IncompatibleIoSelectWithFiberScheduler: # new in 1.21
+  Enabled: true
+Lint/LambdaWithoutLiteralBlock: # new in 1.8
+  Enabled: true
+Lint/NoReturnInBeginEndBlocks: # new in 1.2
+  Enabled: true
+Lint/NonAtomicFileOperation: # new in 1.31
+  # Disable until I understand the cop's implications - jdc 2022-07-14
+  # https://docs.rubocop.org/rubocop/1.31/cops_lint.html#lintnonatomicfileoperation
```

Comment:
Here are the offenses:
>app/models/language_tracking.rb:97:28: W: [Correctable] Lint/NonAtomicFileOperation: Remove unnecessary existence checks File.exist?.
    FileUtils.mkpath(path) unless File.exist?(path)
                           ^^^^^^^^^^^^^^^^^^^^^^^^
test/general_extensions.rb:478:40: W: [Correctable] Lint/NonAtomicFileOperation: Remove unnecessary existence checks File.exist?.
        File.delete("#{filename}.old") if File.exist?("#{filename}.old")
                                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
test/general_extensions.rb:488:33: W: [Correctable] Lint/NonAtomicFileOperation: Remove unnecessary existence checks File.exist?.
      File.delete(new_filename) if File.exist?(new_filename)
                                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
test/models/ip_stats_test.rb:115:35: W: [Correctable] Lint/NonAtomicFileOperation: Remove unnecessary existence checks File.exist?.
    File.delete(MO.ip_stats_file) if File.exist?(MO.ip_stats_file)
                                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
test/models/script_test.rb:115:31: W: [Correctable] Lint/NonAtomicFileOperation: Remove unnecessary existence checks File.exist?.
    FileUtils.rm(output_file) if File.exist?(output_file)
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^